packages/instrumentation-socket.io/src/socket.io.ts

```diff
@@ -0,0 +1,170 @@
+import {
+    context,
+    setSpan,
+    Span,
+    SpanKind,
+    SpanStatusCode,
+    getSpan,
+    diag,
+    suppressInstrumentation,
+} from '@opentelemetry/api';
+import {
+    InstrumentationBase,
+    InstrumentationNodeModuleFile,
+    InstrumentationModuleDefinition,
+    InstrumentationNodeModuleDefinition,
+    isWrapped,
+} from '@opentelemetry/instrumentation';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import io from 'socket.io';
+import { SocketIoInstrumentationConfig } from './types';
+import { VERSION } from './version';
+
+export class SocketIoInstrumentation extends InstrumentationBase<typeof io> {
+    static readonly component = 'socket.io';
+    protected _config!: SocketIoInstrumentationConfig;
+    private moduleVersion: string;
```

Comment:
this method is deprecated, as it is possible to have multiple installations of the library with different versions in the same app.
Our new styling is to pass the module version via hooks.

please see how it is done here: https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-ioredis